src/unit_tests/shared/test_enrollment_op.c

```diff
@@ -337,6 +359,31 @@ void test_w_enrollment_concat_group(void **state) {
     w_enrollment_concat_group(buf, group);
     assert_string_equal(buf, " G:'EXAMPLE_GROUP'");
 }
+/**********************************************/
+/************* w_enrollment_concat_key ****************/
+void test_w_enrollment_concat_key_empty_buff(void **state) {
+    expect_assert_failure(w_enrollment_concat_key(NULL, (keyentry *)1));
+}
+
+void test_w_enrollment_concat_key_empty_key_structure(void **state) {
+    char *buf = *state;
+    expect_assert_failure(w_enrollment_concat_key(buf, NULL));
+}
+
+void test_w_enrollment_concat_key(void **state) {
+    char *buf = *state;
+    keyentry *keys = NULL;
```

Comment:
We can use the keyentry_init() and keyentry_free() methods suggested before.

Reply:
Done